expression/scalar_function.go

```diff
@@ -83,11 +83,12 @@ func NewFunction(ctx context.Context, funcName string, retType *types.FieldType,
 	if builtinRetTp := f.getRetTp(); builtinRetTp.Tp != mysql.TypeUnspecified {
 		retType = builtinRetTp
 	}
-	return &ScalarFunction{
+	sf := &ScalarFunction{
 		FuncName: model.NewCIStr(funcName),
 		RetType:  retType,
 		Function: f,
-	}, nil
+	}
+	return FoldConstant(sf), nil
```

Comment:
If we fold Constant in here, so  maybe no needs to use the recursive function FoldConstant, because the scalarFunction will have only one level here, use (PlainFoldConstant)[https://github.com/pingcap/tidb/blob/d6e6c25b449eca48490d69750bc5e3f7acaf5ec7/expression/constant_fold.go#L51] is better? @hanfei1991